be/src/cache/mem_space_monitor.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cache/mem_space_monitor.h"
+
+#include "common/config.h"
+#include "runtime/exec_env.h"
+#include "runtime/mem_tracker.h"
+#include "storage/page_cache.h"
+#include "util/gc_helper.h"
+#include "util/thread.h"
+
+#define SLEEP_IN_BG_WORKER(seconds)                                         \
+    int64_t left_seconds = (seconds);                                       \
+    while (!_stopped.load(std::memory_order_consume) && left_seconds > 0) { \
+        sleep(1);                                                           \
+        --left_seconds;                                                     \
+    }                                                                       \
+    if (_stopped.load(std::memory_order_consume)) {                         \
+        break;                                                              \
+    }
+
+namespace starrocks {
+
+void MemSpaceMonitor::start() {
+    _adjust_datacache_thread = std::thread([this] { _adjust_datacache_callback(); });
+    Thread::set_thread_name(_adjust_datacache_thread, "adjust_datacache");
+}
```

Comment:
Use different thread names in `MemSpaceMonitor` and `DiskSpaceMonitor`? like `adjust_datacache_mem_space` and `adjust_datacache_disk_space`

Reply:
Fixed